internal/vshard/cluster.go

```diff
@@ -99,9 +99,11 @@ type Cluster struct {
 
 	mutex  sync.RWMutex
 	logger zerolog.Logger
+
+	onClusterDiscoveredCB func(string, int64, []byte)
 }
 
-func NewCluster(name string, cfg config.ClusterConfig) *Cluster {
+func NewCluster(name string, onClusterDiscovered func(string, int64, []byte), cfg config.ClusterConfig) *Cluster {
```

Comment:
Callbacks should be setting by setters. Do not extend the constructor - it is an optional feature.